### PR TITLE
Archive.org plugin: do not propose archival of private notes

### LINKED ISSUE #637: archive.org: do not propose archival of private notes
Private notes are local resources that are not publicly reachable (by opposition to private bookmarked links that can point to public URLs), so the archive.org icon should not be displayed for them

### PR DESCRIPTION
**Pull request by teromene**
Fixes #637

### CODE REVIEW

File: tests/plugins/PluginArchiveorgTest.php [hunk 1]
```diff
@@ -25,24 +25,49 @@ function setUp()
      */
     function testArchiveorgLinklist()
     {
-        $str = 'http://randomstr.com/test';
+        $str1 = 'http://randomstr.com/test';
+        $str2 = 'http://shaarli.test/?aaaaaa';
+        $str2_real_url = '?aaaaaa';
+
```
Comment (ArthurHoaro) [inline tests/plugins/PluginArchiveorgTest.php]: `$camelCase`

File: tests/plugins/PluginArchiveorgTest.php [hunk 2]
```diff
@@ -25,24 +25,49 @@ function setUp()
...
+                    'url' => $str1,
+                    'private' => 0,
+                    'real_url' => $str1
+                ),
+                array(
+                    'url' => $str2,
+                    'private' => 0,
+                    'real_url' => $str2_real_url
+                ),
+				array(
```
Comment (ArthurHoaro) [inline tests/plugins/PluginArchiveorgTest.php]: Indentation: please use 4 spaces.

**Comment by virtualtam** [comment]
Thanks @teromene :)
Could you please [squash your commits](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#Squashing-Commits) once the reviews have been taken into account?

**Comment by teromene** [comment]
Changed according to review and squashed.

**Comment by virtualtam** [comment]
Cool, I'll test it in the evening

**Comment by ArthurHoaro** [comment]
Thanks!

File: plugins/archiveorg/archiveorg.php [hunk 3]
```diff
@@ -17,8 +17,10 @@ function hook_archiveorg_render_linklist($data)
     $archive_html = file_get_contents(PluginManager::$PLUGINS_PATH . '/archiveorg/archiveorg.html');
 
     foreach ($data['links'] as &$value) {
-        $archive = sprintf($archive_html, $value['url']);
-        $value['link_plugin'][] = $archive;
+        if(!($value['private'] && preg_match('/^\?[a-zA-Z0-9-_@]{6}($|&|#)/', $value['real_url']))) {
```
Comment (virtualtam) [inline plugins/archiveorg/archiveorg.php]: more explicit / fail-fast version:
```php
foreach ($data['links'] as &$value) {
    if($value['private'] && preg_match('/^\?[a-zA-Z0-9-_@]{6}($|&|#)/', $value['real_url'])) {
        continue;
    }
    $archive = sprintf($archive_html, $value['url']);
    $value['link_plugin'][] = $archive;
}
```

File: tests/plugins/PluginArchiveorgTest.php [hunk 4]
```diff
@@ -25,24 +25,49 @@ function setUp()
      */
     function testArchiveorgLinklist()
     {
-        $str = 'http://randomstr.com/test';
+        $str1 = 'http://randomstr.com/test';
+        $str2 = 'http://shaarli.test/?aaaaaa';
```
Comment (virtualtam) [inline tests/plugins/PluginArchiveorgTest.php]: this new tests should go into new test cases (methods), so we have:
- a test covering the nominal case, with "ideal" data
- a (new) test covering Shaarli note filtering, with differently formatted note permalinks (e.g. `?4GOGQA`, `?-LYvXg`, etc.)

File: tests/plugins/PluginArchiveorgTest.php [hunk 5]
```diff
@@ -25,24 +25,49 @@ function setUp()
...
-        $this->assertEquals($str, $link['url']);
+        $this->assertEquals($str1, $data['title']);
+        $this->assertEquals($str1, $link['url']);
 
         // plugin data
         $this->assertEquals(1, count($link['link_plugin']));
-        $this->assertNotFalse(strpos($link['link_plugin'][0], $str));
+        $this->assertNotFalse(strpos($link['link_plugin'][0], $str1));
+
+        //Second link : internal public link, plugin datas should be here
```
Comment (virtualtam) [inline tests/plugins/PluginArchiveorgTest.php]: typos:
- missing space after `//`
- no space before `:`
- `data` is the plural form of `datum`, so no final `s` here :)

File: tests/plugins/PluginArchiveorgTest.php [hunk 6]
```diff
@@ -25,24 +25,49 @@ function setUp()
...
+        $link = $data['links'][1];
+        $this->assertEquals(1, count($link['link_plugin']));
+        $this->assertNotFalse(strpos($link['link_plugin'][0], $str2));
+
+        //Third link : internal private link, plugin datas shouldn't be here
```
Comment (virtualtam) [inline tests/plugins/PluginArchiveorgTest.php]: see previous comment :)

**Comment by teromene** [comment]
Updated PR

**Comment by virtualtam** [comment]
Thanks @teromene  :)

**Pull request merged**